### PR TITLE
feat(edgeswitch): allow to differentiate between running and startup config

### PR DESCRIPTION
**Pull request by mwiora**
##### SUMMARY
Allow the collection of running and startup config to compare these to versions or to save one of them

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
edgeswitch

### CODE REVIEW

**Comment by felixfontein** [comment]
Your change breaks backwards compatibility. I don't think that's a good idea.

**Comment by mwiora** [comment]
> Your change breaks backwards compatibility. I don't think that's a good idea.

oh that's interesting - you mean an earlier version did not recognize the command `show startup-config`?

Just hoping that the most of the ubiquiti edgeswitch users will use the latest stable released and longtermin supported version - which version are we talking about? I didn't find a test for that.

In my case I needed a procedure to stop the processing of changes when there are changes not commited - as it might result in an unstable environment. I think I am not the only one having this requirement.

**Comment by felixfontein** [comment]
No, but you are returning results with a different name. Playbooks/roles expecting the data to be there as `ansible_net_config` will no longer work, since it is now called `ansible_net_runningconfig`. Also, if they use the `gather_subset` parameter, it will no longer work as expected since you renamed the value there.

**Comment by mwiora** [comment]
oh nice. Thanks for that hint!
Agreeing that the runningconfig is config (as it was before) I"ll update my PR, so the only thing that changes is that startupconfig is available too.

File: plugins/modules/network/edgeswitch/edgeswitch_facts.py [hunk 1]
```diff
@@ -69,6 +70,11 @@
   type: str
 
 # config
+ansible_net_startupconfig:
+  description: The startup config from the device
+  returned: when config is configured
+  type: str
```
Comment (felixfontein) [inline plugins/modules/network/edgeswitch/edgeswitch_facts.py:76]:
```suggestion
  type: str
  version_added: 1.2.0
```

**Review by felixfontein — COMMENTED** [review]
Looks good, though I have no idea about edgeswitch, so I cannot say whether this change actually works :)

**Comment by felixfontein** [comment]
@f-bor as the author of this module, could you please take a look?

@mwiora you also need a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to).

**Comment by mwiora** [comment]
thanks man! Next time I'll provide that - I have some more needs on this module, so there will come more most probably ;)

**Comment by f-bor** [comment]
@felixfontein it looks good to me
@mwiora you also need to update the unit tests file ([test_edgeswitch_facts.py](https://github.com/ansible-collections/community.network/blob/main/tests/unit/plugins/modules/network/edgeswitch/test_edgeswitch_facts.py)). test_edgeswitch_facts_default gathers all the facts, except config and interfaces. You also have to exclude startup-config in set_module_args (or it will fail because you did not provide a file with the "show startup-config" output).

**Comment by felixfontein** [comment]
Also check out the linting errors in CI: https://app.shippable.com/github/ansible-collections/community.network/runs/527/1/tests

**Comment by mwiora** [comment]
@felixfontein I hope I didn't miss anything :) checks look good. Just tell me, if you need something from me :)

**Comment by felixfontein** [comment]
@mwiora you still need to add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to). Besides that, looks good from my side.

@f-bor are you still happy with the change?

**Comment by mwiora** [comment]
@felixfontein done :)

**Comment by f-bor** [comment]
@felixfontein still good to me

File: changelogs/fragments/105_edgeswitch_add-startupconfig.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - edgeswitch_facts - Added `startupconfig` to facts module - to allow the comparision between startup and running config
```
Comment (felixfontein) [inline changelogs/fragments/105_edgeswitch_add-startupconfig.yml:2]:
```suggestion
  - edgeswitch_facts - added ``startupconfig`` to facts module - to allow the comparision between startup and running config (https://github.com/ansible-collections/community.network/pull/105).
```

**Comment by felixfontein** [comment]
@mwiora thanks a lot for implementing this!
@f-bor thanks a lot for reviewing!

**Pull request merged**